src/modules/gravityCompensator/main.cpp

```diff
@@ -227,12 +227,31 @@ class gravityModuleCompensator: public RFModule
 
         //------------SPECIAL PARAM TP DEFINE THE HEAD TYPE-----//
         version_tag icub_type;
-        if (rf.check("headV2"))
+        // set icub_type.head_version int to 1 (default) or 2
+        if (rf.check("head_version"))
         {
-            yInfo("'headV2' option found. Using icubV2 head kinematics.\n");
-            icub_type.head_version = 2;
+            if (rf.find("head_version").asDouble()==1.0)
+            {
+                yInfo("\"head_version\" %g requested => Using V1 head kinematics.\n",
+                      rf.find("head_version").asDouble());
+            }
+            else if (rf.find("head_version").asDouble()==2.0 ||
+                rf.find("head_version").asDouble()==2.5)
+            {
+                yInfo("\"head_version\" %g requested => Using V2 head kinematics.\n",
+                      rf.find("head_version").asDouble());
+                icub_type.head_version = 2;
```

Comment:
Ah, so it's only cosmetic... no way then 👎 😉 

I'd definitely prefer to have a clear distinction within app files and command line options until this upgrade is solved once for all.